worlds/saving_princess/Rules.py

```diff
@@ -0,0 +1,120 @@
+from BaseClasses import CollectionState, Location, Entrance
+from worlds.generic.Rules import set_rule
+
+from .Constants import *
+
+
+def set_rules(world: "SavingPrincessWorld"):
```

Comment:
Do you have to import `SavingPrincessWorld` then?

Reply:
Not sure if you mean I'm missing the import (I think this is what other worlds do/what I'm missing):
```py
from typing import TYPE_CHECKING
if TYPE_CHECKING:
  from . import SavingPrincessWorld
```
Or if you are asking why I'm using it in the first place?

Reply:
Yeah, I'm saying you'd want to include that exact code here